### DISCUSSION TITLE
Dashboard with Docker Images

### DISCUSSION THREAD

**Discussion by annabeer** [NVIDIA/NVFlare#1930] [Q&A]
Where can I find documentations on how to get the real-world federated learning working with **only** docker images and the dashboard UI. If I am the Project Admin what should I specify for the docker image location. I can get it working with downloading nvflare on each of the VMs.  Thank you in advance!

**Comment by YuanTingHsieh** [comment]
Hi @annabeer have you checked the information on our documentation site: https://nvflare.readthedocs.io/en/main/user_guide/dashboard_ui.html 

Feel free to leave any questions here so we can clarify and enhance our docs!

**Reply by annabeer** [reply]
Thanks for the offer. Once I download the flare console in the readme.txt it says that I should pip install NVFlare however I would like to do that in an isolated way and immediately make use of Docker. Thanks for your help!

**Reply by YuanTingHsieh** [reply]
@annabeer correct me if I understand wrong,

You want to download the packages from DASHBOARD and then after that, use something like `docker run -it --rm -v [package_path] nvflare_docker_image [some commands]`?

For example for admin:
`docker run -it --rm -v /home/myhome/admin@nvidia.com:/nvflare_packages/admin@nvidia.com [nvflare_docker_image] bash /nvflare_packages/admin@nvidia.com/startup/fl_admin.sh`?

**Comment by chesterxgchen** [comment]
@annabeer  if I understand your question correctly, you wish to get the docker image (only) without additional pip install nvflare ? in other words, you would like to have a docker image with nvflare pre-installed ?  

If that's what you are asking, then we don't have one. As the NVFLARE doesn't have a pre-build docker image (except for Dashboard). user usually need to install additional dependencies (such as Tensorflow, Pytorch etc. and many other python dependencies) for the docker image to be useful. so you can easily build a docker image with nvflare + all these dependencies, and publish them to either docker hub or private docker registry such as company hosted artifactory. then you can specify the docker image location for user to download

**Comment by annabeer** [comment]
Yes perfect thanks!

**Comment by annabeer** [comment]
Thanks again. One more questions. I managed to get the containers running however when I submit the hello-pt example I cannot write the data into the containers in the data folder:

"/dev/root on /workspace type ext4 (rw,relatime,discard)
/dev/root on /data type ext4 (ro,relatime,discard)
/dev/root on /etc/passwd type ext4 (rw,relatime,discard)
/dev/root on /etc/group type ext4 (rw,relatime,discard)"

Is this an error on my side and would you know how to fix it. I somehow cant get my head around how to chance the permission of the data folder inside of the container. Thanks in advance

**Reply by chesterxgchen** [reply]
This is specific to the docker. you can start the docker instance as root with -u 0 and then you can have all the power inside the docker 

docker exec -it -u 0 <docker instance> bash 
https://www.baeldung.com/ops/root-user-password-docker-container
